adafruit_tinylora/adafruit_tinylora_encryption.py

```diff
@@ -10,14 +10,45 @@
 Message Integrity checks.
 * Author(s): adafruit
 """
+
+try:  # typing
+    from typing import Annotated, List, Tuple, TypeAlias
+
+    bytearray2: TypeAlias = Annotated[bytearray, 2]
+    bytearray4: TypeAlias = Annotated[bytearray, 4]
+    bytearray16: TypeAlias = Annotated[bytearray, 16]
+
+    StateMatrix: TypeAlias = List[bytearray4]
+except ImportError:
+    pass
+
+
 # from http://cs.ucsb.edu/~koc/cs178/projects/JT/aes.c
-def xtime(col):
+def xtime(col: int) -> int:
     """xtime impl. for _mix_single_column()"""
     return (((col << 1) ^ 0x1B) & 0xFF) if (col & 0x80) else (col << 1)
 
 
 # AES S-box
-S_BOX = (
+# NOTE(typing): Each of these is 16b, can use bytearray16 here
+S_BOX: Tuple[
+    bytes,
+    bytes,
+    bytes,
+    bytes,
+    bytes,
+    bytes,
+    bytes,
+    bytes,
+    bytes,
+    bytes,
+    bytes,
+    bytes,
+    bytes,
+    bytes,
+    bytes,
+    bytes,
+] = (
```

Comment:
@FoamyGuy what are your thoughts on this, which is the true, correct definition, vs `Tuple[bytes, ...]` which is more readable but would need a docstring to specify that it expects 16 `bytes` objects?

Reply:
If mypy supported the syntax, this ideally would be something like
```
S_BOX: Tuple[bytearray16 * 16]
```

Reply:
I think I'm in favor of `Tuple[bytes, ...]` and the docstring specifying 16 `byte`s. 

If they were listed out like this It would be be somewhat cumbersome to count them by hand for someone trying to use it anyhow so having it mentioned in the docstring would be appreciated I think. And I think in this instance it's worth it to save the space in the code and sacrifice being 100% accurate on the number of them